Lutris v0.5.10

mmstick opened the pull request.

ahmed-al-balochi committed:
added feature: you can now list,install,uninstall a runner using cli

ahmed-al-balochi committed:
new commands to list, install, remove a runner.

ahmed-al-balochi committed:
minor fix to the list runner option

ahmed-al-balochi committed:
renamed functions

ahmed-al-balochi committed:
renamed install_runner_cli to prepare_runner_cli

ahmed-al-balochi committed:
formating fixes after running make sc

ahmed-al-balochi committed:
added wine runner install,remove,list options

ahmed-al-balochi committed:
fixing the Review mistakes in application.py and runner.py

ahmed-al-balochi committed:
added the else statement to wine_runner_uninstall

ahmed-al-balochi committed:
removed lutris -w instead use lutris -r to install wine runner. and minor changes to readme file and runner.py

ahmed-al-balochi committed:
Fixed wine if-statement

Co-authored-by: Alexandru Rudi <alexander.ravenheart@gmail.com>

ahmed-al-balochi committed:
fixed yes if-statement

Co-authored-by: Alexandru Rudi <alexander.ravenheart@gmail.com>

ahmed-al-balochi committed:
if no if-statement

Co-authored-by: Alexandru Rudi <alexander.ravenheart@gmail.com>

ahmed-al-balochi committed:
fix if statement for x86_64 suffix

Co-authored-by: Alexandru Rudi <alexander.ravenheart@gmail.com>

ahmed-al-balochi committed:
add if statement for x86_64 suffix

Co-authored-by: Alexandru Rudi <alexander.ravenheart@gmail.com>

ahmed-al-balochi committed:
fix print statement in runner.py

Co-authored-by: Alexandru Rudi <alexander.ravenheart@gmail.com>

ahmed-al-balochi committed:
fixes to the suggested changes

ahmed-al-balochi committed:
fix try-except statements

Co-authored-by: Alexandru Rudi <alexander.ravenheart@gmail.com>

ahmed-al-balochi committed:
updated runner.uninstall()

Co-authored-by: Alexandru Rudi <alexander.ravenheart@gmail.com>

ahmed-al-balochi committed:
migrated the methods from runner.py to application.py

ahmed-al-balochi committed:
resolved required changes plus minor adjustments

ahmed-al-balochi committed:
included the InvalidRunner into try-except

danieljohnson2 committed:
Fallback to a 'package' icon for sidebar runner

This is what the preferences dialog does; the sidebar does the same.

ahmed-al-balochi committed:
fixed the long print statement

ahmed-al-balochi committed:
fixed the long print statement

ahmed-al-balochi committed:
sorted the runnersNames in alphabetical order

ahmed-al-balochi committed:
added whitespace around ':' operator

ahmed-al-balochi committed:
removed whitespace before ':' operator

danieljohnson2 committed:
Add a duplicate game command

The game copies the configuration of the old one to a new ID and gets
a new name (with a number on the end).

The duplicate does not copy the game files; the idea is to create an
alternate configuration for the same game install.

danieljohnson2 committed:
Unhook the emission hooks of a game-bar

Otherwise, the game bar keeps handling these signals forever!

danieljohnson2 committed:
Remove extra logging

danieljohnson2 committed:
More linty goodness

Remove leftover import

danieljohnson2 committed:
If the stop command can't stop a process with SIGTERM try SIGKILL

This gives processes 1 seconds to die, then the hammer falls.

danieljohnson2 committed:
Convert PIDS to int

It's quite surprising that the PIDs are sometimes strings; let's convert
the PIDs that came from /proc into ints.

danieljohnson2 committed:
Longer, asynchronous wait for death

This makes it more complex, but it will now wait up to 5 seconds for
SIGTERM to work.

To make that tolerable, it waits (and periodically checks the processes)
on a thread, so the UI does not freeze up.

ElektroJake committed:
Updated the German translation

strycore committed:
Load lutris media like a normal service

strycore committed:
Improve Origin logging

strycore committed:
Merge pull request #4044 from danieljohnson2/dj/missing_gog_id_handling

Correct handling for missing GOG id

strycore committed:
Add check that db_game is set before launching an install

strycore committed:
Install Ubisoft Connect client if not already installed

strycore committed:
Disable resolution switching options on Wayland

strycore committed:
Handle lack of AppID on Steam games

strycore committed:
Use a date format that is more consistent across systems

strycore committed:
Allow setting a return code for a task that won't exit with 0

strycore committed:
Document return code option

strycore committed:
Avoid crash if Steam client isn't installed

strycore committed:
Only show Ubisoft games for PC

strycore committed:
Add return_code option to execute command

danieljohnson2 committed:
The game-update signal may add a game to the Lutris window

There seem to be several code paths that expect this, so let's do it.

The PR still tries to update or remove the game as appropriate, but if
this fails it will go ahead and update the entire view. This approach
should ensure that any filters that were applied still are.

strycore committed:
Allow overriding runner, runtime and cache folders from settings file

strycore committed:
Add option to configure user directory in Dolphin

strycore committed:
Don't raise an error when no files can't be retrieved from GOG

strycore committed:
Don't use MonitoredCommand for synchronous prelaunch script

strycore committed:
Add i18n to protected folders

strycore committed:
Update Windows folder names to new standard

strycore committed:
Document launch_configs

strycore committed:
Put name of currently playing game in now-playing.txt

strycore committed:
Put service actions in a drop down

strycore committed:
Revert "Put service actions in a drop down"

This reverts commit a97545abe9da0f17cc6781fdd1eabcd1bcb09c31.

strycore committed:
Add button to run service launchers

strycore committed:
Fix crash when a GOG game has no DLC

danieljohnson2 committed:
Disconnect duplicate games from any service

Lutris expects each service game to have at most one PGA game.

danieljohnson2 committed:
Check for the presence "ownerPlatformGroups"

If this is not available, we'll assume that the game is a PC game,
since this seems safer than the opposite.

Layerex committed:
Fix incorrect EasyRPG runner executable path

strycore committed:
This was filtering out extras, not unowned DLCs

strycore committed:
Fixes for DLC extras

danieljohnson2 committed:
Refactor delayed-kill code to allow runner specific behavior

danieljohnson2 committed:
Introduce force-stop method on the runner

Though it just calls right back into the game for now

danieljohnson2 committed:
Override the force-stop method

This will use wineserver -k to kill WINE games 'nicely'.

danieljohnson2 committed:
Remove 'Kill all WINE processes' command

Stop now executes the same function, but with SIGKILL backup.

danieljohnson2 committed:
Correct comment

strycore committed:
Adapt auto-installers for DLC as extras

strycore committed:
Add network tools to PATH

strycore committed:
Add splash screen

rachmadaniHaryono committed:
build: package distro for linux

heidiwenger committed:
Create fi.po

A translation for Finnish language.

heidiwenger committed:
Update fi.po

Few mistakes right away corrected...

bflanagin commented:
- [x] Installs
- [x] launches

Error
```
No such file or directory: '/home/system76/.cache/lutris/coverart'
```
Continues after confirming the error. It will display the error on every launch until the directory is manually created.

mmstick commented:
They fixed it 6 days ago, but perhaps we'll wait for the next release.

strycore committed:
Fix paths for GOG lauch configs

strycore committed:
Better error handling for Origin authentication

strycore committed:
Remove log message for default wine prefix location

strycore committed:
Merge branch 'master' into dark-mode-support

strycore committed:
Merge pull request #4065 from A6GibKm/dark-mode-support

Add support for system dark mode

strycore committed:
Bump min safe version to Wine 7

strycore committed:
Remove duplicate imports

strycore committed:
Merge pull request #4078 from danieljohnson2/dj/stop_wine-ier

Stop harder, but WINE gets wineserver -k not SIGTERM

strycore committed:
Fix XDGMedia

strycore committed:
Remove error message on setting not found

strycore committed:
Remove 'Runners:' text and ErrorDialog from CLI options

strycore committed:
Add export feature

strycore committed:
Remove handler for -d flag

strycore committed:
Add import function

strycore committed:
Use safe_dump for YAML saving

SwimmingTiger committed:
Remove unnecessary execute permission on po/de.po

SwimmingTiger committed:
update POTFILES

SwimmingTiger committed:
update zh_CN.po for translation

SwimmingTiger committed:
update Simplified Chinese translation for 0.5.10

SwimmingTiger committed:
Fix wrong wine version being used to create prefix during prelaunch phase

The wine_path parameter should be passed to the create_prefix function, otherwise the highest installed wine version will be used instead of the version selected by the user.

danieljohnson2 committed:
Just use default styling for the sidebar

This should work with any theme.

strycore committed:
Make LibGnomeDesktop optional

strycore committed:
Remove gir1.2-gnomedesktop requirement

strycore committed:
Disable gamescope on nvidia

bexnoss committed:
Display GOG DLC extras during base game install

michaelarnauts committed:
Fix install from local script

strycore committed:
Add entries to changelog

strycore committed:
Suppress pylint warnings

strycore committed:
contentstatsid key might not be present

strycore committed:
Fix typo

danieljohnson2 committed:
Update the game store immediately

Don't wait for idle time anymore.

danieljohnson2 committed:
Align the splash screen to the upper left

stephanlachnit committed:
debian: compress with gzip

Signed-off-by: Stephan Lachnit <stephanlachnit@debian.org>

keenanweaver committed:
Move subtitles to runner options

keenanweaver committed:
Add scale-factor

keenanweaver committed:
Add opl-driver

keenanweaver committed:
Add output-rate

keenanweaver committed:
Add music-driver

keenanweaver committed:
Add multi-midi

keenanweaver committed:
Add native-mt32 + fixes

keenanweaver committed:
Add enable-gs

keenanweaver committed:
Add filtering

keenanweaver committed:
Add joystick

keenanweaver committed:
Add language

keenanweaver committed:
Add midi-gain, music-volume and fixes

keenanweaver committed:
Fix

keenanweaver committed:
Add platform + fixes

keenanweaver committed:
Add render-mode, soundfont, and fixes

danieljohnson2 committed:
Search for Lutris games asynchronously

Use AsynCall to query the Lutris api without freezing the UI.

Also, add a Gtk.Spinner that shows when the search is happening.

Also, use a SearchEntry for searching.

kabinbr1 committed:
(i18n) Update pt_BR translation

Full Portuguese (Brazil) translation

kabinbr1 committed:
(i18n) Fixes for pt_BR

kabinbr1 committed:
(i18n) pt_BR fixes 2

kabinbr1 committed:
(i18n) pt_BR fixes

keenanweaver committed:
Typos

kabinbr1 committed:
Update pt_BR.po

keenanweaver committed:
Hide scale-factor and fixes

keenanweaver committed:
Add remaining options

keenanweaver committed:
Fix talk speed

keenanweaver committed:
Rearrange entries

keenanweaver committed:
Correct mistakes

keenanweaver committed:
Separate strings

kabinbr1 committed:
Update pt_BR.po

kabinbr1 committed:
Update pt_BR.po

Woops

danieljohnson2 committed:
Add option to limit number of cores

And that number might be 2! or 7!

danieljohnson2 committed:
Integrate the single_cpu setting with the new limit

This way, the old setting is still honored, and and is used to turn on
the limit- which defaults to 1 core now.

THe only downside I can see is that the setting name, 'single_cpu'
is not longer quite right.

danieljohnson2 committed:
Improved messages for the options

mackal committed:
platform_system needed "==" not just "="

Vistaus committed:
Updated Dutch translation

A6GibKm committed:
Use GtkFileChooserNative

This uses the system native file chooser using portals.

- Use .show() instead of .run() when we make use of the "response"
signal so we do not block the main thread.
- The native file chooser uses the accept response
- The dialog has to be destroyed rather than hidden.
- Folders operate the same as files, so we use get_filename instead of
get_current_folder which returns None with the native file chooser

Fixes: https://github.com/lutris/lutris/issues/2924

A6GibKm committed:
Port FileDialog to GtkFileChooserNative

We cannot subclass FileChooserNative since it is a final class.

A6GibKm committed:
Port DirectoryDialog to GtkFileChooserNative

We cannot subclass FileChooserNative since it is a final class.

Folders operate the same as files, so the correct api is get_filename,
get_current_folder will return None.

A6GibKm committed:
FileChooserNative: Hold a reference to the dialog

The native filechooser is not a widget managed by gtk as it is handled
by the system, if we do not keep a strong reference to it on the
"response" signal, the app crashes.

A6GibKm committed:
notifications: Use GNotifications

Instead of libnotify. This is more sandbox friendly and does not
require an additional library.

We also add X-GNOME-UsesNotifications=true to the desktop file, so that
GNOME Software can pick up this info.

strycore committed:
Remove platform requirement for distro

strycore committed:
Remove version requirement from distro

strycore committed:
Remove python-magic dependency

strycore committed:
Enable more services by default

strycore committed:
Simplify play command for scummvm

telanus committed:
Add support for 86Box 

Add basic support for 86Box. It launches Config files, config files need o be created by hand atm.

telanus committed:
fixed a copy and Paste error

strycore committed:
Increase timeout for prefix creation since Wine builds have gone obese

A6GibKm committed:
contextual_menu: Use popup_at_pointer

gtk_menu_popup is deprecated since 3.22 and causes the following error on wayland:

    Gdk-Message: 22:29:12.612: Window 0x220c760 is a temporary window without parent, application will not be able to position it on screen.
    (lutris:883897): Gdk-CRITICAL **: 22:29:12.624: gdk_wayland_window_handle_configure_popup: assertion 'impl->transient_for' failed

A6GibKm committed:
poetry: Update python version to 3.6.2

Black has a hard requirement on python>=3.6.2.

A6GibKm committed:
css: Use a color that behaves better with themes

Using currentColor on backgrounds allows for a style that works both in Adwaita:light
and Adwaita:dark, and as higher chances of working on an arbitrary theme.

cazzoo committed:
Revamped screen (using listbox) and added subscreen


Fixing pulse when extracting runner

danieljohnson2 committed:
Prevent the creation of a blank game on GOG update install

The install save() method must provide a game_id (to be launched), so I
add code to find one- though maybe it is not always the right one.

Also, I've added code to suppress the effort to resave the game after
installing should there be no game ID.

strycore committed:
Adjust some text labels

danieljohnson2 committed:
Handle None for the COL_USAGE

We'll treat that as 0 and that should produce a reasonable result.

danieljohnson2 committed:
Include the app_count in all "View game" buttons

It crashes for me when it tries to use the % operator with no
placeholder in the format string. So let's get one in there.

kabinbr1 committed:
Update pt_BR.po

Getting the latest changes in master and also some other fixes, some "woops!"

A6GibKm committed:
Use Gdk.BUTTON_SECONDARY rather than hardcoded val

danieljohnson2 committed:
Raise RuntimeError when the Ubisoft service returns an error response

This way we don't wind up with a worse error downstream, and the message they provide may
be helpful.

Also, some defensive code to try to avoid reading an auth response that
isn't really an auth response (in case an error gets though somehow)

danieljohnson2 committed:
Wrap the exception to explain better

The user may need to log in again, so we'll suggest something in that
direction.

A6GibKm committed:
Set valign on button

Button appear as expanded, in many cases buttons require valign=CENTER
when in a listboxrow.

GloriousEggroll committed:
util.wine.wine.py: fix missing _version_number typo in is_version_fsync()

GloriousEggroll committed:
util.wine.wine.py: just check entire version string in is_version_fsync() and is_version_esync()

We can just check the entire version string rather than trying to check individual parts from parse_version.

This fixes newer lutris-ge-proton7-* builds from incorrectly being detected as not having fsync enabled.

The cause is due to parse_version failing to parse the version string correctly since lutrig-ge-proton7-* naming scheme has changed in recent versions.

A6GibKm committed:
Use GTK ScreenSaver inhibit API when possible

GTK comes with an implementation of ScreenSaver interface which will
either try to use the org.freedesktop.ScreenSaver or
org.gnome.ScreenSaver DBus interfaces when running on the host, or the
org.freedesktop.portal.ScreenSaver interface when running inside a
sandbox. Note that sandboxes only have access to the .portal variant of
the interface.

To make use of this API, one has to set the
GtkApplication:register-session property to True on construction.

The MATE and XFCE desktop environments will still use their custom
implementations of the ScreenSaver interface as they seem to not support
the Freedesktop implementation in their current releases.

Another benefit is that GTK's api is not blocking.

To make matters worse, the Inhibit and Uninhibit methods used previosly
might not be exposed to PyGObject even if the DBus interface is.

See:
- https://flatpak.github.io/xdg-desktop-portal/#gdbus-org.freedesktop.portal.Inhibit
- https://docs.gtk.org/gtk3/method.Application.inhibit.html

strycore committed:
Only show splash screen when on main games section

strycore committed:
Cleanup

strycore committed:
Merge branch 'master' into lutris_to_steam

strycore committed:
Merge pull request #2729 from buxel/lutris_to_steam

Add Lutris games as steam shortcuts

strycore committed:
Refactor / cleanup vdf related code

strycore committed:
Fix tests

liberavia committed:
Initial implementation of steam shortcuts

tannisroot committed:
Pass new DXVK env variable to enable NVAPI

tannisroot committed:
Add EAC option

liberavia committed:
Only create shortcuts for non-steam-games

liberavia committed:
Add used runner in shortcut name

liberavia committed:
Remove unused imports

liberavia committed:
Remove unused function

liberavia committed:
Determine need in todo

liberavia committed:
Make sure mandatory gameid exists

liberavia committed:
Make sure steam games won't be added as shortcut

liberavia committed:
Optimized imports

kabinbr1 committed:
Update pt_BR.po

Ubisoft messages translated
Easy Anti-Cheat message translated
Steam for windows translation fix (the "for Windows"/"para Windows" part was missing)
"Itch not implemented" - translated
And other small fixes

liberavia committed:
Get arround steam 9-digit userid

liberavia committed:
Get arround steam 9-digit userid

liberavia committed:
Remove coverart due it cannot be assigned to shortcut

1reverseengineer committed:
Added check for NoneType in prefix.py

1reverseengineer committed:
Due to failed PR check -> Replaced == with is on line 65

tannisroot committed:
Enable anticheat options by default

kabinbr1 committed:
Merge branch 'lutris:master' into master

1reverseengineer committed:
Updated due to recommendation in PR-discussion

- Removed parentheses around if-statement (line 65)
- Replaced logger.error with logger.warning

strycore committed:
Disable splash screen

liberavia committed:
Add appid

liberavia committed:
Add vdf filecheck

liberavia committed:
Make sure steam shortcuts will only be offered if available

liberavia committed:
Handle multiple shortcut vdf files

mpetuska committed:
install script json schema

mpetuska committed:
install script json schema

mpetuska committed:
install script json schema

mpetuska committed:
install script json schema

mpetuska committed:
install script json schema

mpetuska committed:
install script json schema

liberavia committed:
Add artwork

mpetuska committed:
install script json schema

liberavia committed:
Fix response of vdf_file_exists()

liberavia committed:
Fix comments

liberavia committed:
Use Move functionality to resources

danieljohnson2 committed:
More robuse game-error signal handling

It will now deal with any exception, and will be handled directly by
LutrisWindow so it won't matter which codepath tries  to start a game.

danieljohnson2 committed:
Raise game config errors for various things

Instead of using ErrorDialog in get_gameplay_info, raise the exception
that seems intended.

This way, the error dialog displayed can have a parent. And when used
from the command line you get console output.

danieljohnson2 committed:
This error message should be localized

Shouldn't it?

danieljohnson2 committed:
Correct doc-comment, fix pylint complaints

danieljohnson2 committed:
We need to signal the game has stopped

If not, it remains in the 'launching' state

strycore committed:
Show installer credits in installer picker

danieljohnson2 committed:
Add cancellation to the LaunchConfigSelectDialog

Both an actual Cancel button and implicit cancellation if you just
close the window.

strycore committed:
Remove Humble Bundle from services enabled by default until fixed

strycore committed:
Merge pull request #4151 from liberavia/add_to_steamui

Implementation of steam shortcuts

strycore committed:
Remove unneeded list comprehension

strycore committed:
Merge pull request #4170 from danieljohnson2/dj/localize_xephyr

The 'Xephyr not found' error message should be localized

strycore committed:
Merge pull request #4153 from kassindornelles/master

Update pt_BR.po

strycore committed:
Merge pull request #4168 from mpetuska/master

Installer script  YAML schema

strycore committed:
Merge pull request #4164 from 18iteration/master

Add NoneType check for variable folder in util/wine/prefix.py

strycore committed:
Remove runner name from Steam shortcut

strycore committed:
Merge pull request #4176 from danieljohnson2/dj/cancel_config_select

Add cancellation to the LaunchConfigSelectDialog

danieljohnson2 committed:
Tolerate ints as accepted_return_code

We'll just turn whatever we get  into a string.

strycore committed:
Remove steam_tag variable

strycore committed:
Replace Dolphin cache version exception with a warning

strycore committed:
Merge branch 'master' into dj/raising_errors

strycore committed:
Merge pull request #4171 from danieljohnson2/dj/raising_errors

More robust error handling for game configuration issues

strycore committed:
Remove merge marker

strycore committed:
Merge pull request #4179 from danieljohnson2/dj/int_return_code

Tolerate ints as accepted_return_code

strycore committed:
Simplify Mangohud option

strycore committed:
Print out failed process on FileNotFound errors

strycore committed:
Add label for extras selection screen

strycore committed:
Update Pilllow version

strycore committed:
Remove Python 3.6 support

strycore committed:
Remove uninstalled games with no playtime on startup

strycore committed:
Avoid a crash when no DB game is returned

strycore committed:
Fix script executable not being returned

strycore committed:
Fix pylint warnings

strycore committed:
Complete changelog

jacobgkau reviewed (approved):
Installs/updates successfully, launches on 22.04, 21.10, and 20.04.
